source/Calamari/Deployment/PackageRetention/Repositories/JsonJournalRepository.cs

```diff
@@ -92,5 +85,10 @@ void Save()
                 fileSystem.OverwriteAndDelete(journalPath, tempFilePath);
```

Comment:
Do we still need the semaphore in `Save`?

Reply:
Nope!